components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3675,4 +3675,8 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     }
 
     async getIDToken(): Promise<void> {}
+    public async resolveContext(ctx: TraceContextWithSpan, contextUrl: string): Promise<WorkspaceContext> {
+        const user = this.checkAndBlockUser("resolveContext");
```

Comment:
What's the access profile here? Do we need to authorize this call to ensure you only resolve something you have access to? How does this work for private repositories?

Reply:
FWIW, I think this would be a good case for a service - ContextService

Reply:
It uses your GH/GL/BB token, so you can't access anything else

Reply:
> FWIW, I think this would be a good case for a service - ContextService

We can create a "ContextService" once we are able to implement a proto service here.

Reply:
Where's the GH/GL/BB token stored? From what I can see, we're not passing the request/connection credentials down into these calls. Tracing it further, it looks like we store these and then retrieve based on the user (ID)